### ISSUE TITLE
Enumerating ErrorTrigger tracefilter events throws 'collection was modified' exception

### ISSUE THREAD

**Issue by jr01** [Azure/azure-webjobs-sdk-extensions#176]
When processing multiple errors in parallel an exception can be thrown when enumerating the tracefilter events collection. 

```
public static void ErrorTrigger([ErrorTrigger] TraceFilter filter, TextWriter log)
        {
            foreach (var evt in filter.Events)
            {
                // Do some work that takes some time
               Thread.Sleep(100);
            }
        }
```

> An exception of type 'System.InvalidOperationException' occurred in mscorlib.dll but was not handled in user code
> Additional information: Collection was modified; enumeration operation may not execute.

#### Repro steps

1. Download my repo https://github.com/jr01/WebJobsErrorTriggerIssue
2. Fill in App.config connectionstrings.
2. F5/Run

#### Expected behavior

No exception thrown.

#### Actual behavior

After some seconds the mentioned exception is thrown

#### Known workarounds

Copy the collection to a new array and use the new array

```
var newArray = (TraceEvent[])Array.CreateInstance(typeof(TraceEvent), filter.Events.Count);
filter.Events.CopyTo(newArray, 0);
            foreach (var evt in newArray)
            {
                // Do some work that takes some time
               Thread.Sleep(100);
            }
```

#### Related information 

- The exception can be reproduced also using a timertrigger. I haven't tried others.
- Might be related to https://github.com/Azure/azure-webjobs-sdk-extensions/issues/166

**Comment by mathewc** [comment]
For now, a workaround would be for you to take a snapshot before enumeration by doing `filter.Events.ToArray()`. The Events collection is a "live" collection, so it appears the issue is that the issue is we're adding new error events to the collection behind the scenes while you're trying to enumerate.

**Comment by jr01** [comment]
Yes ToArray() is a easier than the Array.CreateInstance + filter.Events.CopyTo workaround I came up with :) Thanks @mathewc !

**Comment by soninaren** [comment]
addressed in here #184
